hls-graph/src/Development/IDE/Graph/Internal/Database.hs

```diff
@@ -200,7 +200,9 @@ compute db@Database{..} stack key mode result = do
                     (getResultDepsDefault mempty previousDeps)
                     deps
         _ -> pure ()
-    atomicallyNamed "compute" $ SMap.focus (updateStatus $ Clean res) key databaseValues
+    atomicallyNamed "compute and run hook" $ do
+        runHook
+        SMap.focus (updateStatus $ Clean res) key databaseValues
```

Comment:
Even though these are in the same STM transaction, I think it probably makes sense to do this first? Potentially the hook could look at this state? I'm not sure.